modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -190,7 +190,8 @@ const moveSelectionToMarker = (editor: Editor, marker: HTMLElement | null): void
       rng = nextRng;
       dom.remove(parentBlock);
     } else {
-      dom.add(parentBlock, dom.create('br', { 'data-mce-bogus': '1' }));
+      // If parentBlock is a table cell, add a br without 'data-mce-bogus' attribute.
+      dom.add(parentBlock, dom.create('br', isTableCell(parentBlock) ? {} : { 'data-mce-bogus': '1' }));
```

Comment:
The `isTableCell` check is done in the above `if` statement as well. Might be worth having a variable like `const blockIsTableCell = isTableCell(parentBlock);` that can be used in both places